pkg/probe/exec/exec.go

```diff
@@ -66,7 +67,8 @@ func (pr execProber) Probe(e exec.Cmd) (probe.Result, string, error) {
 			if exit.ExitStatus() == 0 {
 				return probe.Success, string(data), nil
 			}
-			return probe.Failure, string(data), nil
+			probeFailedMsg := fmt.Sprintf("exitCode: %d, stderr: %s, stdout: %s", exit.ExitStatus(), exit.Error(), string(data))
```

Comment:
`data` here includes both stdout and stderr (lines 55 & 56 above). From the implementation, it looks like `exit.Error` will typically be a summary of why the process halted, and the exit status, making it redundant with the exit code. A better message might be to just append `exit.Error()` to the `string(data)`. E.g:

```suggestion
			probeFailedMsg := string(data)
			if probeFailedMsg[len(probeFailedMsg)-1] != '\n' {
			  probeFailedMsg += '\n'
			}
			probeFailedMsg += exit.Error()
```

Reply:
@tallclair appreciate for your reviews. your suggestion is excellent, I have updated the code. 

Here is a small change, that exitCode could not appear in exit.Error(), thus I updated method exit.String() to return both exitCode & exitErrorMsg.

please take a look, thanks a lot.